README.md

```diff
@@ -1 +1,157 @@
-This is the J.I. provided GPIO device service.
+# GPIO Device Service
+## Overview
+GPIO Micro Service - device service for connecting GPIO devices to EdgeX.
+
+- Function:
+  - This device service use sysfs to control GPIO devices. For a connected GPIO device, export the needed pin number, set correct GPIO direction, and then start to read or write data from GPIO device.
+- Physical interface: system gpio (/sys/class/gpio)
+- Driver protocol: IO
+
+
+
+## Usage
+- This Device Service runs with other EdgeX Core Services, such as Core Metadata, Core Data, and Core Command.
+- The gpio device service can contains many pre-defined devices which were defined by `configuration.toml`, such as `Custom-GPIO-Device`. These devices are created by the gpio device service in core metadata when the service first initializes. 
+- Each pre-defined device has its own profile, such as `device-custom-gpio-profile.yaml`, which choose the sysfs gpios to export by `deviceResources`. Then it add `coreCommands`  and `coreCommands` with the  name same to `deviceResources`.
+- After the  gpio device service has started,  we can read or write the system gpio corresponding to pre-defined device by specific command.
+
+```yaml
+deviceResources:
+  -
+    name: "Power"
+    description: "system power gpio export"
+    attributes: { number: "/134" }
+    properties:
+      value:
+        { type: "Bool", readWrite: "RW" }
+
+  -
+    name: "Led"
+    description: "system led gpio export"
+    attributes: { number: "/65" }
+    properties:
+      value:
+        { type: "Bool", readWrite: "W" }
+
+  -
+    name: "Button"
+    description: "system button gpio export"
+    attributes: { number: "/66" }
+    properties:
+      value:
+        { type: "Bool", readWrite: "R" }
+```
+
+- Since GPIO sysfs interface is deprecated after Linux version 4.8, we provide two ABI interfaces: the sysfs version and the new chardev version. Please change GOTAG in Makefile, and select the interface you like. For the sysfs version, in device resource, you need to provide attribute number like  "/123", which 123 is the pin number you selected. As for chardev version, the number should be like "0/17", which the first number is for gpiochip, and the second number is for selected line number. 
```

Comment:
Maybe we can have a discussion on this topic, since currently the device profile supports multiple devices, they may share the same gpiochip, so, maybe we can to put this into device attributes?